docs/concepts/gas.md

```diff
@@ -0,0 +1,200 @@
+
+This page covers these dynamics in more detail, starting with an an excerpt from the [NEAR Whitepaper](https://nearprotocol.com/papers/the-official-near-white-paper/) below:
+
+<blockquote class="info">
+<strong>NEAR Platform Economics -- What is Gas?</strong><br><br>
+
+Fundamentally, the NEAR platform is a marketplace between willing participants.
+
+On the supply side, operators of the validator nodes and other fundamental infrastructure need to be incentivized to provide these services which make up the community cloud.  On the demand side, the developers and end-users of the platform who are paying for its use need to be able to do so in a way which is simple, clear and consistent so it helps them.
+
+A blockchain-based cloud provides several specific resources to the applications which run atop it:
+
+- **Compute (CPU)**: This is the actual computer processing (and immediately available RAM) which run the code in a contract.
+- **Bandwidth ("Network")**: This is the network traffic between participants and users, including messages which submit transactions and those which propagate blocks.
+- **Storage**: Permanent data storage on the chain, typically expressed as a function of both storage space and time.   
+ 
+Existing blockchains like Ethereum account for all of these in a single up front transaction fee which represents a separate accounting for each of them but ultimately charges developers or users only once in a single fee.  
+
+**This is a high volatility fee commonly denominated in "gas".**
+
+Developers prefer predictable pricing so they can budget and provide prices for their end users.  The pricing for the above-mentioned resources on NEAR is an amount which is slowly adjusted based on system usage (and subject to the smoothing effect of resharding for extreme usage) rather than being fully auction-based. This means that a developer can more predictably know that the cost of running transactions or maintaining their storage.
+
+Initially, all of these resources will be priced and paid in terms of NEAR tokens. In the future, they may also be priced in terms of a stable currency denomination (for example a token pegged to the $USD).
+
+[read more here](https://nearprotocol.com/papers/the-official-near-white-paper/#economy)
+
+</blockquote>
+
+## How do I buy gas?
+
+You don't. 
+
+Accounts purchase gas automatically using NEAR tokens and at various points in time including:
+
+1. **when a transaction is processed** (ie. a function call is made, an account is created, tokens are transferred, etc) \
+    The amount of gas used by a transaction depends on the transaction itself and is fixed at genesis (ie. it requires 1 unit of gas to store 1 byte of information and it requires 3 units of gas to compute 2+2).  The actual *cost of this gas*, it's price in NEAR tokens is calculated dynamically and on an ongoing basis based on system load.
```

Comment:
@nearmax Could you provide a more realistic example of how much storing / computing may cost?

@amgando I think the sentence "ie. it requires 1 unit of gas to store 1 byte" needs to be reworded to reflect that it only serves an example, not the actual exact numbers.

Reply:
We shouldn't be giving the actual numbers, since we might not be event in the right order of magnitude.

Reply:
for sure, i can reword that as follows:

>  The amount of gas used by a transaction depends on the transaction itself and is fixed at genesis (ie. through some governance process, we may decide that it requires 1 unit of gas to store 1 byte of information but it requires 3 units of gas to compute the addition of two numbers, 3+2 -- note this is just a toy example, actual gas required for storage and compute can be verified experimentally using examples like those below).

but i'm still struggling to build my own confidence around the complete costs of all possible storage, compute and networking events in the system.  i think i recall someone was working on this, a kind of cost estimator, but i haven't seen anything about it.

i can see in my local `genesis.json` file that there are many settings for various bits about gas (`min_gas_price` and `max_gas_burnt` in general per block?, `runtime_config` has a dozen details, `wasm_config` has another dozen details, etc)

Reply:
ok, took a closer look at the numbers in `genesis.json` and i see how any specific numbers can be wildly misleading.  i'll replace with this: 

> The amount of gas used by a transaction depends on the details of the transaction itself.  At genesis, the blockchain is configured with many different cost parameters (in units of gas).  Through some governance process, these parameters may be changed over time.  The actual *cost of this gas*, its price in NEAR tokens, is calculated dynamically and on an ongoing basis based on system load, inflation and other factors.

is this a better representation, @nearmax and @frol?

Reply:
Maybe worth mentioning that transactions other than function call will cost a fixed amount of gas (unless we change the protocol).